sui_core/src/authority.rs

```diff
@@ -67,6 +70,95 @@ pub mod authority_notifier;
+    pub shared_obj_tx: IntCounter,
+    tx_already_processed: IntCounter,
+    num_input_objs: Histogram,
+    num_shared_objects: Histogram,
+    batch_size: Histogram,
+}
+
+// Override default Prom buckets for positive numbers in 0-50k range
+const POSITIVE_INT_BUCKETS: &[f64] = &[
+    1., 2., 5., 10., 20., 50., 100., 200., 500., 1000., 2000., 5000., 10000., 20000., 50000.,
+];
+
+impl AuthorityMetrics {
+    pub fn new() -> AuthorityMetrics {
+        Self {
+            tx_orders: register_int_counter!(
+                "total_transaction_orders",
+                "Total number of transaction orders"
+            )
+            .unwrap(),
+            total_certs: register_int_counter!(
+                "total_transaction_certificates",
+                "Total number of transaction certificates handled"
+            )
+            .unwrap(),
+            // total_effects == total transactions finished
+            total_effects: register_int_counter!(
+                "total_transaction_effects",
+                "Total number of transaction effects produced"
+            )
+            .unwrap(),
+            total_events: register_int_counter!("total_events", "Total number of events produced")
+                .unwrap(),
+            signature_errors: register_int_counter!(
+                "total_signature_errors",
+                "Number of transaction signature errors"
+            )
+            .unwrap(),
+            shared_obj_tx: register_int_counter!(
+                "num_shared_obj_tx",
+                "Number of transactions involving shared objects"
+            )
+            .unwrap(),
+            tx_already_processed: register_int_counter!(
+                "num_tx_already_processed",
+                "Number of transaction orders already processed previously"
+            )
+            .unwrap(),
+            num_input_objs: register_histogram!(
+                "num_input_objects",
+                "Distribution of number of input TX objects per TX",
+                POSITIVE_INT_BUCKETS.to_vec()
+            )
+            .unwrap(),
+            num_shared_objects: register_histogram!(
+                "num_shared_objects",
+                "Number of shared input objects per TX",
+                POSITIVE_INT_BUCKETS.to_vec()
+            )
+            .unwrap(),
+            batch_size: register_histogram!(
+                "batch_size",
+                "Distribution of size of transaction batch",
+                POSITIVE_INT_BUCKETS.to_vec()
+            )
+            .unwrap(),
+        }
+    }
+}
+
+impl Default for AuthorityMetrics {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+// One cannot register a metric multiple times.  We protect initialization with lazy_static
+// for cases such as local tests or "sui start" which starts multiple authorities in one process.
+lazy_static! {
+    static ref METRICS: Arc<AuthorityMetrics> = Arc::new(AuthorityMetrics::new());
+}
```

Comment:
use `once-cell` instead and since this is a static global you do not need to wrap it in a reference counted object (Arc)

Reply:
See my other comments on the Arc, I think its preferable to keep for now